homeassistant/components/adax/__init__.py

```diff
@@ -3,14 +3,18 @@
 from __future__ import annotations
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
 
-PLATFORMS = [Platform.CLIMATE]
+from .climate import AdaxDataHandler
+from .const import PLATFORMS
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Adax from a config entry."""
+    adax_data_handler = AdaxDataHandler(entry, hass)
```

Comment:
Do you need the AdaxDataHandler class?

Could something like this work?


```
        if entry.data.get(CONNECTION_TYPE) == LOCAL:
            hass.data[entry.entry_id] = AdaxLocal(
                entry.data[CONF_IP_ADDRESS],
                entry.data[CONF_TOKEN],
                websession=async_get_clientsession(hass, verify_ssl=False),
            )
        else:
            hass.data[entry.entry_id] = Adax(
                entry.data[ACCOUNT_ID],
                entry.data[CONF_PASSWORD],
                websession=async_get_clientsession(hass),
            )
            ```

Reply:
Yes but then we would need to get the rooms from the cloud every time. I think it is better to share the data and minimise cloud polling.

Reply:
No, it is cached in the library

Reply:
Doesn't work and I'm not sure why.
Adding the below code in async_setup_entry for both platforms.
```
    adax_data_handler = hass.data[entry.entry_id]
    for room in await adax_data_handler.get_rooms():
        _LOGGER.warning(
            "Found rooms"
        )
```
The platform that is listed first in:
PLATFORMS = [Platform.SENSOR, Platform.CLIMATE]
Will log "Found rooms" for each room.
But the second platform will hang on "await adax_data_handler.get_rooms()"